index.js

```diff
@@ -296,21 +312,64 @@ function createFocusTrap(element, userOptions) {
   // kind of need to capture the action at the keydown phase.
   function checkTab(e) {
     updateTabbableNodes();
-    if (e.shiftKey && e.target === state.firstTabbableNode) {
-      e.preventDefault();
-      tryFocus(state.lastTabbableNode);
-      return;
+    const isShift = e.shiftKey;
+
+    if (isShift) {
+      const startOfGroupIndex = state.tabbableGroups.findIndex(
+        ({ firstTabbableNode }) => e.target === firstTabbableNode
+      );
+
+      const isStartOfAGroup = startOfGroupIndex !== -1;
+      if (isStartOfAGroup) {
```

Comment:
Could we check for `>= 0` instead of strictly looking for a single value? I prefer not to depend on a single value whenever possible. "If the index is any negative value, then it's not the start of a group..."

Also, I don't think we ever use the `isStartOfAGroup` variable anywhere, so why not just `if (startOfGroupIndex >= 0)`?

Reply:
Done.  However, just as a different thought, I actually prefer the `-1` check here.

`-1` has a special meaning here.  It means that it's not found in the array.  When you check against `-1`, it's clear (in my mind) that what you're doing is checking to see if it is part of the array or not.

When you use `>=0`, it takes me a minute longer to see what is going on here.  The `>=0` makes it seem like I'm checking against a range, where the number may be `-2`, `-1`, `-13`, or some positive number.  When really, I'm not checking for a range condition, the only possible negative number here is -1.

I also like naming it, for reasons I stated above.  I guess I took that from Clean Code by Martin.

I only add this as a varying opinion, not at all to banter.  Just spreading ideas.